src/crypto.ts

```diff
@@ -226,37 +229,45 @@ export default class Crypto {
       }
     })
 
-    const downloadPromises : Array<Promise<void>> = []
-    for (let fieldId in attachmentRecords) {
-      // Original name for the file is not found
-      if (filenames[fieldId] === undefined) return null
-
-      downloadPromises.push(
-        axios.get(attachmentRecords[fieldId], { responseType: 'json' })
-      .then((downloadResponse) => {
-        const encryptedAttachment: EncryptedAttachmentContent = downloadResponse.data
-        const encryptedFile: EncryptedFileContent = {
-          submissionPublicKey: encryptedAttachment.encryptedFile.submissionPublicKey,
-          nonce: encryptedAttachment.encryptedFile.nonce,
-          binary: decodeBase64(encryptedAttachment.encryptedFile.binary),
-        }
-
-        return this.decryptFile(formSecretKey, encryptedFile)
-      }).then((decryptedFile) => {
-        if (decryptedFile === null) throw new Error("Attachment decryption failed")
-        decryptedRecords[fieldId] = { filename: filenames[fieldId], content: decryptedFile }
-      }))
+    const fieldIds = Object.keys(attachmentRecords)
+    // Check if all fieldIds are within filenames
+    if (!areFieldsValid(fieldIds, filenames)) {
+      return null
     }
 
     try {
-      await Promise.all(downloadPromises)
-    } catch (e) {
+      await Promise.all(
+        fieldIds.map((fieldId) => {
+          return axios
+            .get(attachmentRecords[fieldId], { responseType: 'json' })
+            .then((downloadResponse) => {
+              const encryptedAttachment: EncryptedAttachmentContent =
+                downloadResponse.data
+              const encryptedFile: EncryptedFileContent = {
+                submissionPublicKey:
+                  encryptedAttachment.encryptedFile.submissionPublicKey,
+                nonce: encryptedAttachment.encryptedFile.nonce,
+                binary: decodeBase64(encryptedAttachment.encryptedFile.binary),
+              }
+              return this.decryptFile(formSecretKey, encryptedFile)
+            })
+            .then((decryptedFile) => {
+              if (decryptedFile === null)
+                throw new Error('Attachment decryption failed')
+              decryptedRecords[fieldId] = {
+                filename: filenames[fieldId],
+                content: decryptedFile,
+              }
+            })
+        })
```

Comment:
I actually prefer the old method of assigning the Promise array to a variable first to avoid so much nesting, wdyt?

Reply:
i don't have a strong opinion on this; main conflict here seems to be nesting vs mutating - i just don't like mutating HAHA but yea, will change accordingly!